db/src/main/java/io/lantern/db/DB.kt

```diff
@@ -943,55 +833,90 @@ class Transaction internal constructor(
                 }
             })
         }
+
+        changesBySubscriber.forEach { (subscriber, changes) ->
+            subscriber.onChanges(changes)
+        }
     }
 }
 
 internal class DetailsSubscriber<T : Any>(
```

Comment:
There were some race conditions in DetailsSubscriber, so I completely changed how it works to take care of them.